packages/jetpack-ai-calypso/src/logo-generator/components/upgrade-screen.tsx

```diff
@@ -10,15 +10,21 @@ import { __ } from '@wordpress/i18n';
 import { EVENT_PLACEMENT_FREE_USER_SCREEN, EVENT_UPGRADE } from '../../constants';
 import type React from 'react';
 
-export const UpgradeScreen: React.FC< { onCancel: () => void; upgradeURL: string } > = ( {
-	onCancel,
-	upgradeURL,
-} ) => {
-	const upgradeMessage = __(
+export const UpgradeScreen: React.FC< {
+	onCancel: () => void;
+	upgradeURL: string;
+	reason: 'feature' | 'requests';
+} > = ( { onCancel, upgradeURL, reason } ) => {
+	const upgradeMessageFeature = __(
 		'Upgrade your Jetpack AI for access to exclusive features, including logo generation. This upgrade will also increase the amount of requests you can use in all AI-powered features.',
 		'jetpack'
 	);
 
+	const upgradeMessageRequests = __(
+		'Upgrade your Jetpack AI to increase the amount of requests you can use in all AI-powered features.',
```

Comment:
Since the user will only see the small modal, it may be worth to explain why they need to upgrade: they don't have enough balance of requests considering the cost of the logo generation.